mlrun/execution.py

```diff
@@ -1090,6 +1091,15 @@ def _write_tmpfile(self, json=None):
                 fp.write(data)
                 fp.close()
 
+    def _cleanse_labels(self):
+        """Remove labels that should not be stored to the corresponding run"""
+        labels = deepcopy(self._labels)
+        # Remove host (pod name) label if it's not a logging worker:
+        if not self.is_logging_worker():
+            labels.pop("host", None)
+
+        return labels
```

Comment:
the cleansing shoudl either take an argument of labels to cleanse, or the function name should be changed to `_resolve_cleansed_labels` or you can `_resolve_labels_for_update()` without feeding it a labels argument

the reason im saying that is because when you want to `_x_something` where x is the action, you would expect to feed the function with an argument as oppose to where x is "resolve" or "get" where you want it to return you something / compile something for you